packages/p2p-v2/src/features/my-profile/screens/MyProfileStats/MyProfileStats.tsx

```diff
@@ -0,0 +1,141 @@
+
+    return (
+        <div className='p2p-v2-my-profile-stats__item'>
+            <span>
+                {label}{' '}
+                <button
+                    className={clsx(
+                        'p2p-v2-my-profile-stats__item--inactive',
+                        !hasClickedLifetime && shouldShowLifetime && 'p2p-v2-my-profile-stats__item--active'
+                    )}
+                    onClick={() => {
+                        setHasClickedLifetime(false);
+                        onClickLifetime?.(false);
+                    }}
+                >
+                    {duration}
+                </button>{' '}
+                {shouldShowLifetime && (
+                    <>
+                        |{' '}
+                        <button
+                            className={clsx(
+                                'p2p-v2-my-profile-stats__item--inactive',
+                                hasClickedLifetime && 'p2p-v2-my-profile-stats__item--active'
+                            )}
+                            onClick={() => {
+                                setHasClickedLifetime(true);
+                                onClickLifetime?.(true);
+                            }}
+                        >
+                            lifetime
+                        </button>
+                    </>
+                )}
+            </span>
+            <span>
+                <strong>
+                    {value} {currency}
+                </strong>
+            </span>
+        </div>
+    );
+};
+
+type MyProfileStatsProps = {
+    advertiserId?: string;
+};
+
+export const MyProfileStats = ({ advertiserId }: MyProfileStatsProps) => {
+    const [shouldShowTradeVolumeLifetime, setShouldShowTradeVolumeLifetime] = useState(false);
+    const [shouldShowTotalOrdersLifetime, setShouldShowTotalOrdersLifetime] = useState(false);
+    const { data } = useAdvertiserStats(advertiserId);
+    const { data: activeAccountData } = useActiveAccount();
+
+    if (!data || !activeAccountData) return <h1>Loading...</h1>;
+
+    // NOTE: This component is only visible for accounts with USD currency
+    if (activeAccountData?.currency !== 'USD') return <></>;
+
+    const {
+        averagePayTime,
+        averageReleaseTime,
+        buyCompletionRate,
+        buyOrdersCount,
+        sellCompletionRate,
+        sellOrdersCount,
+        totalOrders,
+        totalOrdersLifetime,
+        tradePartners,
+        tradeVolume,
+        tradeVolumeLifetime,
+    } = data;
+
+    const getTimeValueText = (minutes: number) => `${minutes === 1 ? '< ' : ''}${minutes} min`;
+    const getCurrencyText = (currency: number) =>
+        new Intl.NumberFormat('en-US', { maximumFractionDigits: 2, minimumFractionDigits: 2, style: 'decimal' }).format(
+            currency
+        );
+
+    return (
+        <div className='p2p-v2-my-profile-stats'>
+            <MyProfileStatsItem
```

Comment:
What about mapping this? so we don't have to repeat MyProfileStatsItem multiple times

Reply:
I think these are distinct though, if I map it I would have to create a mapper object for it, plus its not dynamic and its number of components is always fixed